time.c

```diff
@@ -3348,32 +3348,91 @@ tmcmp(struct tm *a, struct tm *b)
 
 /*
  * call-seq:
- *   Time.utc(year, month = 1, day = 1, hour = 0, min = 0, sec_i = 0, usec = 0) -> new_time
- *   Time.utc(sec_i, min, hour, day, month, year, dummy, dummy, dummy, dummy) -> new_time
+ *   Time.utc(year, month = 1, mday = 1, hour = 0, min = 0, sec = 0, usec = 0) -> new_time
+ *   Time.utc(sec, min, hour, mday, month, year, dummy, *more_dummies)` -> new_time
  *
- * Returns a new \Time object based the on given arguments;
- * its timezone is UTC.
+ * Returns a new \Time object based the on given arguments,
+ * always in the UTC timezone.
  *
- * In the first form (up to seven arguments), argument +year+ is required.
+ * With one to seven arguments given, returns a new \Time object
+ * based on the given values:
  *
- *   Time.utc(2000)                  # => 2000-01-01 00:00:00 UTC
- *   Time.utc(0, 1, 2, 3, 4, 5, 6.5) # => 0000-01-02 03:04:05.0000065 UTC
+ *   Time.utc(2000)  # => 2000-01-01 00:00:00 UTC
+ *   Time.utc(-2000) # => -2000-01-01 00:00:00 UTC
  *
- * In the second form, all ten arguments are required,
- * though the last four are ignored.
- * This form is useful for creating a time from a 10-element array
- * such as is returned by #to_a.
+ * There are no minimum and maximum values for +year+;
+ * otherwise, these are the minimum and maximum values:
  *
- *   array = Time.now.to_a
- *   # => [55, 14, 10, 7, 7, 2022, 4, 188, true, "Central Daylight Time"]
- *   array[5] = 2000
- *   Time.utc(*array) # => 2000-07-07 10:14:55 UTC
+ *   Time.utc(-2000, 1, 1, 0, 0, 0, 0)
+ *   # => -2000-01-01 00:00:00 UTC
+ *   Time.utc(2000, 12, 31, 23, 59, 60, 999999)
+ *   # => 2001-01-01 00:00:00.999999 UTC # Note the rollover caused by 60 sec.
  *
- * Parameters:
- * :include: doc/time/year.rdoc
- * :include: doc/time/mon-min.rdoc
- * :include: doc/time/sec_i.rdoc
- * :include: doc/time/usec.rdoc
+ * The values may be numerics convertible to integers:
+ *
+ *   Time.utc(Float(0.0), Rational(1, 1), 1.0, 0.0, 0.0, 0.0, 0.0)
+ *   # => 0000-01-01 00:00:00 UTC
+ *
+ * Or string integers:
+ *
+ *   a = %w[0 1 1 0 0 0 0 0]
+ *   # => ["0", "1", "1", "0", "0", "0", "0", "0"]
+ *   Time.utc(*a) # => 0000-01-01 00:00:00 UTC
+ *
+ * And the value for +month+ may be a case-insensitive 3-letter month name:
+ *
+ *   a = %w[0 jan 1 0 0 0 0 2]
+ *   # => ["0", "jan", "1", "0", "0", "0", "0", "2"]
+ *   Time.utc(*a) # => 0000-01-01 00:00:00 UTC
+ *   a = %w[0 JAN 1 0 0 0 0 2]
+ *   # => ["0", "JAN", "1", "0", "0", "0", "0", "2"]
+ *   Time.utc(*a) # => 0000-01-01 00:00:00 UTC
+ *
+ * As seen in the above examples, when seven or fewer arguments are given,
+ * the arguments are interpreted as in the first calling sequence above:
+ *
+ *   Time.utc(year, month = 1, mday = 1, hour = 0, min = 0, sec = 0, usec = 0)
+ *
+ * For longer argument lists:
+ *
+ * - When eight arguments are given, the last is ignored.
```

Comment:
Moved to antepenultimate remark, about exceptions (per doc guide), with the assumption that the new exception will be added.

Reply:
I opened #6281 which fixes this inconsistency. The current implementation is also inconsistent with other Ruby implementation since this case is an undefined behaviour.